netbox/project-static/src/util.ts

```diff
@@ -163,17 +163,21 @@ function buildUrl(destination: string): string {
 
   // If the `origin` exists in the API path (as in the case of paginated responses), remove it.
   const origin = new RegExp(window.location.origin, 'g');
-  let path = pathname.replaceAll(origin, '');
+  let path = pathname
+    .replaceAll(origin, '')
+    .split('/')
+    .filter(p => p);
 
   const basePath = getBasePath();
 
-  // If the `BASE_PATH` already exists in the URL, remove it.
-  if (basePath !== '' && path.includes(basePath)) {
-    path = path.replaceAll(basePath, '');
+  // If the `BASE_PATH` already exists in the URL, and it is the first element, remove it.
+  if (basePath !== '' && path[0].includes(basePath)) {
```

Comment:
Like my previous comment though, this should never be necessary. We should be able to assume at this point that the URL being passed has already been processed to accommodate `BASE_PATH` and proceed accordingly.